### PR TITLE
find UUID in completed only after pending/waiting

### PR DESCRIPTION
**Pull request by jspricke**
9fe6215 added completed to the search to allow reopening tasks but the
issue could have been completed multiple times before resulting in an
MultipleMatches exception. This change only considers completed tasks if
the issue is not in the pending/waiting set and also arbitrarily returns
the first found result.

### CODE REVIEW

File: bugwarrior/db.py [hunk 1]
```diff
@@ -197,6 +196,24 @@ def find_taskwarrior_uuid(tw, keys, issue):
...
 
+    # old bugwarrior did not consider completed task so there could be multiple ones matching
+    for service, key_list in six.iteritems(keys):
+        if any([key in issue for key in key_list]):
+            results = tw.filter_tasks({
+                'and': [("%s.is" % key, issue[key]) for key in key_list],
+                'or': [
+                    ('status', 'pending'),
+                    ('status', 'waiting'),
+                    ('status', 'completed')
```
Comment (ryneeverett) [inline bugwarrior/db.py]: Shouldn't we only be checking `completed` issues here? If it were `pending` or `waiting` it would have already been returned.

**Review by ryneeverett — APPROVED** [review]
Thanks for reporting this oversight!

File: bugwarrior/db.py [hunk 2]
```diff
@@ -197,6 +196,19 @@ def find_taskwarrior_uuid(tw, keys, issue):
             )
         )
 
+    # old bugwarrior did not consider completed task so there could be multiple ones matching
+    for service, key_list in six.iteritems(keys):
+        if any([key in issue for key in key_list]):
+            results = tw.filter_tasks({
+                'and': [[("%s.is" % key, issue[key]) for key in key_list], ('status', 'completed')]
```
Comment (ryneeverett) [inline bugwarrior/db.py]: Looks like these lists need to be concatenated, so something like:

```py
               'and': [("%s.is" % key, issue[key]) for key in key_list] + ['status', 'completed'],
```

**Comment by ryneeverett** [comment]
While I find your PR more readable I'm leaning towards #797 for performance reasons. If the extra taskwarrior query only occurred on edge cases I would consider that acceptable but it looks like as it current stands there will be an extra query for every new issue before adding it to the database. Thoughts?

File: bugwarrior/db.py (hunk 2 above)
Comment (jspricke) [reply bugwarrior/db.py]: looks better, thx. pushed.

**Comment by jspricke** [comment]
I agree that #797 is probably the better fix.

**Pull request closed without merging**